src/wallet/wallet.h

```diff
@@ -373,43 +373,13 @@ struct COutputEntry
 class CMerkleTx
 {
 public:
-    CTransactionRef tx;
-    uint256 hashBlock;
-
-    /* An nIndex == -1 means that hashBlock (in nonzero) refers to the earliest
-     * block in the chain we know this or any in-wallet dependency conflicts
-     * with. Older clients interpret nIndex == -1 as unconfirmed for backward
-     * compatibility.
-     */
-    int nIndex;
-
-    CMerkleTx()
-    {
-        SetTx(MakeTransactionRef());
-        Init();
-    }
-
-    explicit CMerkleTx(CTransactionRef arg)
-    {
-        SetTx(std::move(arg));
-        Init();
-    }
-
-    void Init()
-    {
-        hashBlock = uint256();
-        nIndex = -1;
-    }
-
-    void SetTx(CTransactionRef arg)
-    {
-        tx = std::move(arg);
-    }
-
     ADD_SERIALIZE_METHODS;
 
     template <typename Stream, typename Operation>
     inline void SerializationOp(Stream& s, Operation ser_action) {
+        CTransactionRef tx;
```

Comment:
6779b7cbdeec091bf2c5a704c07a018641625d5f

Just noting that it's not really necessary to make these local variables since that unserialized `CMerkleTx` are discarded anyway. I guess you are doing this to underline the idea, so maybe also add a comment here?

Reply:
Yeah, it's equivalent, but I think the intent is clearer this way. Why have data members in a class where any instantiated objects are just going to be thrown away?

I don't think an additional comment is necessary here - the code along with the comment on the `CMerkleTx` class should be clear enough.